src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -204,6 +204,16 @@ const keriNotificationsChangeHandler = async (
     //TODO: Use dispatch here, handle logic for the multisig notification
   } else if (event?.a?.r === NotificationRoute.MultiSigRot) {
     //TODO: Use dispatch here, handle logic for the multisig rotation notification
+  } else if (event?.a?.r === NotificationRoute.GrantRequest) {
+    dispatch(
```

Comment:
We should do some checks here before we put up the request, not after we accept. So call `getKeriExchange` here and check if the tunnel AID is correct.

Then maybe we can also pass the tunnel AID, SAID, and domain URL here to `IncomingRequest` to display it as part of the message that pops up on-screen. All of these elements will be in the message sent from the tunnel by Jaime

Reply:
And please rename it to `getKeriExchangeMessage`

Reply:
`getKeriExchange` is created long ago. If we rename it in this PR and merge it into `feat/tunnel` it might not change the `develop` branch. I think we shouldn't rename it in this PR.

Reply:
ok fair enough